src/ReverseProxy/Configuration/ConfigurationConfigProvider.cs

```diff
@@ -60,7 +57,7 @@ public IProxyConfig GetConfig()
                 _subscription = ChangeToken.OnChange(_configuration.GetReloadToken, UpdateSnapshot);
                 UpdateSnapshot();
             }
-            return _snapshot;
+            return _snapshot!;
         }
 
         private void UpdateSnapshot()
```

Comment:
```suggestion
        [MemberNotNull(nameof(_snapshot))]
        private void UpdateSnapshot()
```

This should allow you to remove the forgiveness operator on line 60.

Reply:
Is there an alternative to MemberNotNull when cross compiled to 3.1?

Reply:
You can compile the attributes in to the assembly. We do this for our ns2.0 projects: https://github.com/dotnet/aspnetcore/blob/main/src/Shared/Nullable/NullableAttributes.cs